lightly/data/_video.py

```diff
@@ -0,0 +1,137 @@
+""" """
+
+#
+# 
+
+import os
+from PIL import Image
+from torchvision import datasets
+
+from torchvision.io import read_video, read_video_timestamps
+
+
+def _video_loader(path, timestamp, pts_unit='sec'):
+    """
+
+    """
+    # random access read from video
+    frame, _, _ = read_video(path,
+                             start_pts=timestamp,
+                             end_pts=timestamp,
+                             pts_unit=pts_unit)
+    # read_video returns tensor of shape 1 x W x H x C
+    frame = frame.squeeze()
+    # convert to PIL image
+    # TODO: can it be on CUDA? -> need to move it to CPU first
+    image = Image.fromarray(frame.numpy())
+
+    return image
+
+
+def _make_dataset(directory, extensions=None, is_valid_file=None, pts_unit='sec'):
+    """
+
+    """
+
+    if extensions is not None:
+        def _is_valid_file(filename):
+            return filename.lower().endswith(extensions)
+
+    # find all instances
+    instances = []
+    for fname in os.listdir(directory):
+
+        if not _is_valid_file(fname):
+            continue
+
+        path = os.path.join(directory, fname)
+        instances.append(path)
+
+    # get timestamps
+    timestamps, fpss = [], []
+    for instance in instances:
+        ts, fps = read_video_timestamps(instance, pts_unit=pts_unit)
+        timestamps.append(ts)
+        fpss.append(fps)
+
+    # get offsets
```

Comment:
not clear to me what offsets are and why we need them, maybe add a small comment to explain